src/pipelines/computePipeline.ts

```diff
@@ -81,15 +80,46 @@ export class ComputeNode<
     this.shouldRecompute = false;
     this._output = NoOutput;
     this.parentNodes = parentNodes;
-    this.computeCallback = this.compute.bind(this);
+
+    // Recompute when a parent node change
     const nParents = this.parentNodes.length;
     for (let parentIndex = 0; parentIndex < nParents; ++parentIndex) {
-      this.parentNodes[parentIndex].onOutputUpdate(this.computeCallback);
+      const parentNode = this.parentNodes[parentIndex];
+      const computeCallback = this.createComputeCallback(parentNode);
+      parentNode.onOutputUpdate(computeCallback);
     }
   }
 
+  /**
+   * Create and return a callback function that will trigger recomputation of this node
+   * @param parentNode The parent node that will trigger a recomputation
+   * @returns The callback that needs to be called when the parentNode's output changes
+   */
+  private createComputeCallback(parentNode: ComputeNode<any, any>) {
+    let previousParentOutput = parentNode.output;
+    const computeCallback = () => {
+      const currentParentOutput = parentNode.output;
+      // Don't recompute when the parent output changes from NoOutput to NoOutput
```

Comment:
if it changes from NoOutput to NoOutput, then, it did not change :)

Reply:
Good point!